slicer_cli_web/models/docker_image.py

```diff
@@ -185,14 +185,16 @@ def saveImage(name, cli_dict, docker_image, user, baseFolder):
         folderModel.setMetadata(tag, labels)
 
         folderModel.setMetadata(tag, dict(slicerCLIType='tag', slicerCLIRestPath=restPath))
-        folderModel.clean(tag)
+        existingItems = {item['name']: item for item in list(folderModel.childItems(tag, filters={'meta.slicerCLIType': 'task'}))}
 
         for cli, desc in six.iteritems(cli_dict):
             item = itemModel.createItem(cli, user, tag, 'Slicer CLI generated CLI command item',
                                         reuseExisting=True)
             itemModel.setMetadata(item, dict(slicerCLIType='task'))
             itemModel.setMetadata(item, desc)
 
+            del existingItems[cli]
```

Comment:
This should be guarded in case the cli is new (tested by manually deleting one task item).